docs/DocGenerator/README.md

```diff
@@ -0,0 +1,205 @@
+# Wazuh QA DocGenerator
+Wazuh - Quality assurance automation self-contained documentation parsing tool
+
+## Rational
+Wazuh QA documentation is designed to be self-contained into the source code of each test.
+
+DocGenerator is the tool in charge of parsing the documentation block from each source code and generate data capable
+to be indexed and displayed.
+
+## Design
+### Input
+DocGenerator parses the information from a QA repository with test files containing a specific comment format:
+
+Each test file has self-contained documentation comment blocks in **YAML** format.
+
+These blocks can contain **Mandatory**, **Optional**, and **Ignored** fields.
+In the [Configuration section](#configuration) it's explained how to configure these fields.
+- **Mandatory** fields must be present in the documentation block and will be dumped into the final
+documentation output.
+- **Optional** fields, if present, will be parsed and dumped into the final documentation output.
+- **Ignored** fields are the ones not listed in the configuration file. They can be added by the developer to explain
+a functionality of the module or a method without transferring this information to the final documentation output.
+
+A test file has a module documentation block at the top level of the file with information related to the complete
+test file. And each test method inside the file will have a test documentation block with the specific information of
+this test.
+
+Additional group information is parsed from the README files found in the repository. Each README file represents
+a group, and every test file at the same or lower folder level than this group is considered as belonging to this group.
+
+Also, groups could be nested and a README found under the level of another group will generate a new group that belongs
+to the first one and contains any group or tests under it.
+
+The specific content of each block is defined in the [Documentation schema section](#documentation-schema)
+
+### Parsing
+Running DocGenerator as specified in the [Usage section](#usage) will scan every test and group file found into the
+include paths of the documentation, it will extract the module and tests blocks from each test file and discard any
+non-documentable field. Also, complementary test-cases information will be extracted from a dry-run of pytest if there
+isn´t a description for them.
+
+### Output
+The parsed and filtered documentation information will be dumped in the Output folder defined by configuration.
```

Comment:
Maybe:  "The parsed and filtered information will be added into the output folder defined in the configuration file.